customize-cra PR review 반영

amorfati0310 opened the pull request:
## 1. PR For What
-지난 PR merge가 되서  리뷰 반영 

## 2. 작업내역 
- [x] 개발 린트 추가 설정 (react-hook, import-plugin, end of line rule 추가)
- [x] 리팩토링 -> packae.json -> devDependencies 
-  [x] api serviceKey나 _return json 같은 기본 parameter 내부적으로 설정

##  3. 이슈 
```js
 "eslint-import-resolver-webpack": "^0.12.1",
  "eslint-plugin-import": "^2.20.1",
```
를 활용해서 `import` 나 `extensions`는 기본 webpack 설정으로  잡는데 
`config-overrides.js`에 추가로 설정한 `alias`는 못 잡고 `eslint rule` 에러가 난다.

amorfati0310 committed:
add: eslint-import-reslover-webpack -> fix import extension

여전히 @ alias 설정은 webpack 설정에서 못 참조를 하는 것 같다.
- jsconfig.json에 @ alias 설정을 넣어줘도 그대로여서 당분간 import/no-unresolved는 off로 진행
- devDependencies 구분되게 수정
[Ticket: 05]

amorfati0310 committed:
fix: Api 기본 params 추가

[Ticket: 05]

amorfati0310 committed:
Add EsLint rule Enforce end of line
[Ticket: 05]

y0c commented on .eslintrc.js:
```diff
@@ -6,17 +6,35 @@ const ERROR = 2;
     'import/no-unresolved': OFF,
-    'import/extensions': OFF,
+    'react-hooks/rules-of-hooks': ERROR, // Checks rules of Hooks
+    'react-hooks/exhaustive-deps': WARNING, // Checks effect dependencies
```
이건 Error가 좋아보여요 꼭 필요할때만 eslint-disable-next-line을

y0c commented on .eslintrc.js:
```diff
@@ -6,17 +6,35 @@ const ERROR = 2;
   },
+  settings: {
+    'import/resolver': {
+      webpack: {
```
`config-overrides` 는 webpack 설정이 아니라서 `eslint-import-resolver-alias` 를 쓰고 map하시면 될거에용

amorfati0310 replied:
@y0c  `eslint-import-resolver-alias` 에 map 설정을 추가 했는데도 못 잡네요 ;ㅁ; 좀 더 알아보겠습니당

y0c replied:
```
alias: {
        map: [
          ['babel-polyfill', 'babel-polyfill/dist/polyfill.min.js'],
          ['helper', './utils/helper'],
          ['material-ui/DatePicker', '../custom/DatePicker'],
          ['material-ui', 'material-ui-ie10']
        ],
        extensions: ['.ts', '.js', '.jsx', '.json']
      }
``` 
예시를 보면 배열이 한번 더 감싸져있네요 👀

amorfati0310 replied:
@y0c 앗 배열 안으로 넣어주니 잘 되네영 감사합니다 :)

y0c commented on src/api/index.js:
```diff
@@ -2,8 +2,22 @@ import axios from 'axios';
 
 const MiseApi = axios.create({
   baseURL: `http://openapi.airkorea.or.kr/openapi/services/rest/UlfptcaAlarmInqireSvc`,
+  params: {
```
여기들어간게 모든요청에 안붙는다면 interceptor같은걸써볼수도있겠네요. 
https://github.com/axios/axios#interceptors

amorfati0310 commented:
@y0c 리뷰 너무  감사합니다 :) ~~ 
수정해서 반영할게요

amorfati0310 committed:
apply to Review: react-hooks/exhaustive -> Error  and axios interceptor
- intsall eslint-import-resolver-alias
- issue -> map alias 적용 안 됨

[Ticket: 05]

amorfati0310 committed:
fix: eslint-import-resolver-alias  map 잘 못 적용한 부분 수정
- map 배열안에 배열로 들어갔어야 했다

[Ticket: 05]
